src/lib/AST.cpp

```diff
@@ -18,6 +18,42 @@ bool variableVal::operator == (const variableVal& rVal) const{
             return (std::get<bool>(value) == std::get<bool>(rVal.value));
         case ReturnType::FUNC:
             return (std::get<std::shared_ptr<Func>>(value)->mForest == std::get<std::shared_ptr<Func>>(rVal.value)->mForest);
+        case ReturnType::ARRAY: {
+            // Check if one variableVal is NOT an array
```

Comment:
we already know both variable are the same type. I think we can do something like we did on line 20 but with arrays